[#42] members 테이블 memberWithdraw 추가

Yo0oN opened the pull request:
회원 탈퇴 시 회원탈퇴여부, 탈퇴일자 컬럼이 추가되어 회원가입 시에도 회원탈퇴여부, 탈퇴일자 컬럼 입력할 수 있도록 추가

Yo0oN committed:
[#42] members 테이블 memberWithdraw 추가

회원 탈퇴 시 회원탈퇴여부, 탈퇴일자 컬럼이 추가되어 회원가입 시에도 회원탈퇴여부, 탈퇴일자 컬럼 입력할 수 있도록 추가

Jane096 reviewed (commented):
`getUserById` 에서도 삭제여부가 `False` 인 사람만 불러오는게 좋을까요??

**수정**
로그인하고 자기 자신꺼 보는건데 필요없겠네요!!^^ 잘못 생각한듯 해요~~

Yo0oN commented:
> `getUserById` 에서도 삭제여부가 `False` 인 사람만 불러오는게 좋을까요??

`getUserById`는 로그인 후 자신이 정보를 보는데 사용되고 있어서 없어도 괜찮을거같습니다.

대신 로그인시 존재하는 회원인지 확인하고, 회원가입시 중복 아이디인지 확인해주는 `isUserIdExist`이 문제인데,
로그인일 경우 탈퇴회원은 로그인을 못하도록 삭제여부가 False인 사람만 불러와야합니다.
하지만 중복아이디인지 확인할 경우 1년간 같은 아이디로는 재가입이 불가능하게 해야하니 삭제여부에 상관없이 전부 불러와야합니다. 

그러니 `isUserIdExist`를 사용하는 부분을 따로 분리하는 작업도 같이 진행하겠습니다.

+아니면 현재는 `userId`를 Primary Key로 사용중이어서 아이디가 중복되는 것을 막기 위해 이미 탈퇴했더라도 탈퇴한 정보가 사라지기 전인 1년간은 같은 아이디로 재가입이 불가능하게 해두었습니다.
그런데 회원번호인 `userNo` 컬럼을 추가하여 Primary Key로 사용하게 된다면 아이디가 아닌 회원번호로 회원을 구분하기 때문에 `isUserIdExist`를 분리하지 않고 사용할 수 있게됩니다. 대신 아이디가 아닌 회원 번호로 회원 관리를 하게 되므로, 아이디를 이용한 부분들을 수정해야 합니다.
이부분은 더 상의해보고 결정합시다.😓

f-lab-dev commented:
이런 구현 내용에 대해서도 서로간의 상의가 있으면 좋을 것 같네요~ 저 없이도 `google meet`으로 회의도 해보시고 구현 방향에 대해 한번 토론해보세요~

f-lab-dev commented on src/main/resources/mapper/MemberMapper.xml:
```diff
@@ -12,7 +12,9 @@
               address,
-              userLevel
+              userLevel,
+              memberWithdraw,
```
이런 경우는 `deleted` 정도로 컬럼 이름을 지어주면 어떨까요?
`탈퇴`는 도메인 관점의 이름이고 `삭제되었다`는 리소스 관점의 표현이니까요

Yo0oN replied:
네! 관련된 memberWithdrawDate도 함께 deletedDate로 수정하겠습니다!

Yo0oN committed:
feat:members 테이블 컬럼명 수정

memberWithdraw와 memberWithdrawDate 대신 리소스 관점에서의 삭제되었다인 deleted와 deletedDate로 수정

[#42]feat:members 테이블 memberWithdraw 추가
